sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java

```diff
@@ -235,13 +237,14 @@ private static String getParameter(@NonNull Sort.Order order) {
 
         final String direction = order.isDescending() ? "DESC" : "ASC";
 
-        return String.format("r.%s %s", order.getProperty(), direction);
+        return String.format("%s.%s %s", tableName, order.getProperty(), direction);
     }
 
-    static String generateQuerySort(@NonNull Sort sort) {
+    static String generateQuerySort(@NonNull Sort sort, @NonNull String inputTableName) {
         if (sort.isUnsorted()) {
             return "";
         }
+        tableName = inputTableName;
```

Comment:
why are we updating `tableName` here?

Reply:
If the use case is to re-use this tableName variable in the `getParameter()` API above, still I would not recommend updating the static variable. Rather passing it through the API makes more sense as a local variable.

Reply:
We are updating here because in the case of annotated queries, the tableName can be anything so we need to set it to whatever was defined in the query. The private variable has to be static since the functions are static.

Reply:
Added new constant variable.